ee/clickhouse/queries/column_optimizer.py

```diff
@@ -96,6 +96,11 @@ def properties_used_in_filter(self) -> Counter[PropertyIdentifier]:
                     (self.filter.breakdown, self.filter.breakdown_type, self.filter.breakdown_group_type_index)
                 ] += 1
 
+            # If we have a breakdowns attribute then make sure we pull in everything we
+            # need to calculate it
+            for breakdown in self.filter.breakdowns or []:
```

Comment:
non-blocking just 🤔

https://github.com/PostHog/posthog/blob/01191c21338cede1cb8e3b5b43998b31351263f3/ee/clickhouse/queries/column_optimizer.py#L87

That conditional branch include group type index when I guess it'll never be present. Which might not be a problem. And is almost identical to the breakdowns branch. I wonder if there's a nice way to unify them...

Reply:
Yeah not sure about that tbh. I imagine at some point this breakdowns would be all we see internally, which could include the "group" stuff as well. And at the rest api boundary we'd unify any helper breakdown related attributes.